File: Java/P007CategoricalCrossEntropyLoss.java
public class P007CategoricalCrossEntropyLoss {
  public static void main(String[] args) {
    double[] softmaxOutput = new double[] { 0.7, 0.1, 0.2 };
    double[] targetOutput = new double[] { 1, 0, 0 };

    double loss = -(
      (Math.log(softmaxOutput[0]) * targetOutput[0]) +
      (Math.log(softmaxOutput[1]) * targetOutput[1]) +
      (Math.log(softmaxOutput[2]) * targetOutput[2])
    );

    System.out.println(loss);

    System.out.println(-Math.log(0.7));
    System.out.println(-Math.log(0.5));
  }
}

File: Go/p007-Categorical-Cross-Entropy-Loss.go
package main

import (
	"fmt"
	"math"

	"gonum.org/v1/gonum/mat"
)

func main() {
	softmax_output := mat.NewVecDense(3, []float64{0.7, 0.1, 0.2})
	target_output := mat.NewVecDense(3, []float64{1.0, 0.0, 0.0})

	var loss float64 = -(math.Log(softmax_output.AtVec(0))*target_output.AtVec(0) +
						 math.Log(softmax_output.AtVec(1))*target_output.AtVec(1) +
						 math.Log(softmax_output.AtVec(2))*target_output.AtVec(2))
	fmt.Printf("%f\n", loss)

	loss = -math.Log(softmax_output.AtVec(0))
	fmt.Printf("%f\n", loss)
}
